src/main/java/org/isf/generaldata/MessageBundle.java

```diff
@@ -42,6 +45,9 @@
 
 	private static ResourceBundle defaultResourceBundle = null;
 
+	private static String formatSpecifier = "%(\\d+\\$)?([-#+ 0,(\\<]*)?(\\d+)?(\\.\\d+)?([tT])?([a-zA-Z%])";
```

Comment:
if the aim is to count the number of arg, wouldn't be simpler this kind of pattern?
![image](https://user-images.githubusercontent.com/2938553/116936616-6241fc80-ac68-11eb-9f2b-493e81756a26.png)

Anyway it seems not working with Arabic...

Reply:
I argued with myself adding the check because (1) it made the routine slower to execute and (2) it would fail when the same substitutable was used more than once in the string [fake example:  `The variable "{0}" is wrong.  "{0}" cannot be blank.`]

Therefore I just removed it.   If they don't match up then the string will look wrong and it will be reported as a bug.

Reply:
simplicity has its convenience ;)
thank a lot @dbmalkovsky